src/btree/bt_cursor.c

```diff
@@ -168,6 +168,40 @@ __cursor_fix_implicit(WT_BTREE *btree, WT_CURSOR_BTREE *cbt)
     return (btree->type == BTREE_COL_FIX && cbt->compare != -1);
 }
 
+/*
+ * __key_within_bounds --
+ *     Determine if a given key is within the bounds set on a cursor.
+ */
+static int
+__key_within_bounds(WT_CURSOR_BTREE *cbt, WT_ITEM *key, bool *key_out_of_bounds)
+{
+    WT_BTREE *btree;
+    WT_CURSOR *cursor;
+    WT_SESSION_IMPL *session;
+
+    session = CUR2S(cbt);
+    cursor = &cbt->iface;
+    btree = CUR2BT(cbt);
+    *key_out_of_bounds = false;
+
+    WT_ASSERT(session, WT_CURSOR_HAS_BOUNDS(cursor));
+    WT_ASSERT(session, key != NULL);
+
```

Comment:
Could we move the btree->type == BTREE_COL check here, and return ENOTSUP first rather than checking the bounds and then checking for row/col store twice?

Reply:
35425a8bb